packages/block-library/src/navigation/edit/index.js

```diff
@@ -591,8 +591,7 @@ function Navigation( {
 						{ isResponsive && (
 							<>
 								<Button
-									// TODO: Switch to `true` (40px size) if possible
-									__next40pxDefaultSize={ false }
+									__next40pxDefaultSize
```

Comment:
This can be seen on the inspector controls, when we add navigation block and we try to set overlay menu icon, refer to below screenshots.

| Before | After     |
| :-------- | :------- |
| <img width="263" alt="navigation-overlay-before" src="https://github.com/user-attachments/assets/18b7e6bf-2d1b-4ab6-babe-0f589d9493dd"> | <img width="263" alt="navigation-overlay-after" src="https://github.com/user-attachments/assets/0a44307d-e1bb-49fb-a622-1921289ff86b"> |

Reply:
> We're going to need to preserve this height override. I think we'll need to add an !important here.

Thanks for this, would add `!important` to override the `40px` style.

Reply:
It's a tricky balance.

I lean on the side of not decreasing the specificity, especially since it's going to be removed soon anyway.

> If custom Button height like this is a legitimate design requirement, maybe it should be supported officially?

I'm not sure that adding "one more variant" is a scalable solution.

Seeing how many heavily tweaked instances of `Button` we're encountering, I would consider (as already discussed in the past) to introduce an "unstyled" variant of `Button`, which would still benefit from the component's functionality (ie. tooltips, link vs button, focus styles, accessible when disabled, etc) but without any extra cosmetic changes applied (ie. no padding, margin, heighgits, etc).

Alternatively, we could expose a `useButton` hook which does the same thing, and use if with vanilla `<button />` elements.